### PR TITLE
Events

### PR DESCRIPTION
**Pull request by ciliakas**
Added very basic events, both custom and standard to User Database, they can be subscribed to by any class that uses it.
Also removed Data Manager from Goal.cshtml.cs

### CODE REVIEW

File: src/Pages/Goals.cshtml.cs [hunk 1]
```diff
@@ -29,17 +30,20 @@ public class GoalsModel : PageModel
...
 
         public async Task OnGet()
         {
             UserId = int.Parse(User.FindFirst(ClaimTypes.Name).Value);
-            var dataManager = new DataManager(UserId);
             Goals = await GoalDatabase.ReadListAsync(UserId);
-            Savings = dataManager.Income.EntryList.GetSum() - dataManager.Expenses.EntryList.GetSum();
+            var incomes = await EntryDatabase.ReadListAsync(UserId, EntryType.Income);
+            var expenses = await EntryDatabase.ReadListAsync(UserId, EntryType.Expense);
+            Savings = incomes.GetSum() - expenses.GetSum();
```
Comment (Stokalas) [inline src/Pages/Goals.cshtml.cs:51]: This is where the dataManager was removed in goals :D

File: src/Pages/Entry.cshtml.cs [hunk 2]
```diff
@@ -14,17 +15,28 @@ namespace ePiggyWeb.Pages
...
+            GoalDatabase = goalDatabase;
+            UserDatabase = userDatabase;
+            EmailSender = emailSender;
+            UserDatabase.Deleted += OnDeleteUser;
+        }
+
+        private async void OnDeleteUser(object sender, UserModel user)
+        {
+            await EmailSender.SendFarewellEmailAsync(user.Email);
         }
```
Comment (Stokalas) [inline src/Pages/Entry.cshtml.cs:35]: As I undertand we need delete user button somewhere?

File: src/Pages/Login.cshtml.cs [hunk 3]
```diff
@@ -27,9 +29,17 @@ public class LoginModel : PageModel
...
+        public LoginModel(UserDatabase userDatabase, EmailSender emailSender)
         {
             UserDatabase = userDatabase;
+            EmailSender = emailSender;
+            UserDatabase.Registered += OnRegister;
+        }
+
+        private async void OnRegister(object sender, UserModel user)
+        {
+            await EmailSender.SendGreetingEmailAsync(user.Email);
```
Comment (Stokalas) [inline src/Pages/Login.cshtml.cs]: Why is greetings email in login? :D

File: src/Pages/Register.cshtml.cs [hunk 4]
```diff
@@ -36,6 +37,12 @@ public RegisterModel(UserDatabase userDatabase, EmailSender emailSender)
         {
             UserDatabase = userDatabase;
             EmailSender = emailSender;
+            UserDatabase.Registered += OnRegister;
+        }
+
+        private async void OnRegister(object sender, UserModel user)
+        {
+            await EmailSender.SendGreetingEmailAsync(user.Email);
         }
```
Comment (Stokalas) [inline src/Pages/Register.cshtml.cs:47]: Seems neat to send the greetings email in the first place, second nice that events were used :D

**Review by Stokalas — COMMENTED** [review]
I did not test whether or not this works, but I'm happy that events were used, left few questions, will approve when answered. 🙂

File: src/Pages/Login.cshtml.cs (hunk 3 above)
Comment (ciliakas) [reply src/Pages/Login.cshtml.cs]: I have no idea :D Changed it to be On Login, even though it doesn't do anything, just to show that we could do something :D

File: src/Pages/Entry.cshtml.cs (hunk 2 above)
Comment (ciliakas) [reply src/Pages/Entry.cshtml.cs:35]: That would be nice to have, maybe on the header or footer for now, just to show that we have this sort of functionality

File: src/Pages/Goals.cshtml.cs (hunk 1 above)
Comment (ciliakas) [reply src/Pages/Goals.cshtml.cs:51]: Now it makes sense :D

**Review by Stokalas — APPROVED** [review]

**Pull request merged**